app/controllers/characters_controller.rb

```diff
@@ -69,6 +69,22 @@ def update
     end
   end
 
+  def duplicate
+    Character.transaction do
+      @dup = @character.dup
+      @dup.galleries = @character.galleries
+      @character.aliases.find_each do |calias|
+        dupalias = calias.dup
+        @dup.aliases << dupalias
+        dupalias.save
+      end
+      @dup.save
+    end
+
+    flash[:success] = "Character duplicated successfully."
+    redirect_to edit_character_path(@dup)
```

Comment:
I think it's required if I want to then reference it in tests? (unless I get all characters before and after the duplication, find the one new character)